src/sql/workbench/contrib/notebook/test/browser/notebookActions.test.ts

```diff
@@ -259,22 +260,91 @@ suite('Notebook Actions', function (): void {
+				metadata: { language: 'python' },
+				execution_count: 1
+			}],
+			metadata: {
+				kernelspec: {
+					name: 'python',
+					language: 'python',
+					display_name: 'Python 3'
+				}
+			},
+			nbformat: 4,
+			nbformat_minor: 5
+		};
+
 		let mockNotification = TypeMoq.Mock.ofType<INotificationService>(TestNotificationService);
 		mockNotification.setup(n => n.notify(TypeMoq.It.isAny()));
 		let quickInputService = new MockQuickInputService;
+		let mockNotebookModel = new NotebookModelStub(undefined, undefined, testContents);
 
 		let action = new RunParametersAction('TestId', true, testUri, quickInputService, mockNotebookService.object, mockNotification.object);
 
-		// Normal use case
 		const testCells = [<ICellModel>{
 			isParameter: true,
 			source: ['x=2.0\n', 'y=5.0']
 		}];
-
+		mockNotebookEditor.setup(x => x.model).returns(() => mockNotebookModel);
 		mockNotebookEditor.setup(x => x.cells).returns(() => testCells);
 
-		assert.doesNotThrow(() => action.run(testUri));
+		// Run Parameters Action
+		await action.run(testUri);
+		let testShowOptions = {
+			initialContent: JSON.stringify(mockNotebookModel.toJSON()),
+			preserveFocus: true
+		};
+		assert.call(mockNotebookService.object.openNotebook(testUri, testShowOptions), 'Should Open Parameterized Notebook');
+	});
+
+	test('Run with Parameters Action with no parameters in notebook', async function (): Promise<void> {
+		const testContents: azdata.nb.INotebookContents = {
+			cells: [{
+				cell_type: CellTypes.Code,
+				source: [''],
+				metadata: { language: 'python' },
+				execution_count: 1
+			}],
+			metadata: {
+				kernelspec: {
+					name: 'python',
+					language: 'python',
+					display_name: 'Python 3'
+				}
+			},
+			nbformat: 4,
+			nbformat_minor: 5
+		};
+		let expectedMsg: string = localize('noParametersCell', "This notebook cannot run with parameters until a parameter cell is added. [Learn more](https://docs.microsoft.com/sql/azure-data-studio/notebooks/notebooks-parameterization).");
+		let notification = {
+			severity: Severity.Info,
+			message: expectedMsg,
+		};
+
+		let actualMsg: string;
+		let mockNotification = TypeMoq.Mock.ofType<INotificationService>(TestNotificationService);
+		mockNotification.setup(n => n.notify(TypeMoq.It.isAny())).returns(notification => {
+			actualMsg = notification.message;
+			return undefined;
+		});
+		let quickInputService = new MockQuickInputService;
+		let mockNotebookModel = new NotebookModelStub(undefined, undefined, testContents);
+
+		let action = new RunParametersAction('TestId', true, testUri, quickInputService, mockNotebookService.object, mockNotification.object);
+
+		mockNotebookEditor.setup(x => x.model).returns(() => mockNotebookModel);
+
+		// Run Parameters Action
+		await action.run(testUri);
+
+		assert.ok(actualMsg !== undefined, 'Should have received a notification');
+		assert.call(mockNotification.object.notify(notification), 'Should notify user there is no parameter cell');
```

Comment:
And when you do this instead of having a callback which stores the notification.message you can just verify that it was actually called with the value you expect.

For example here you can see that I'm verifying that login was called with a specific password as its 3rd parameter : https://github.com/microsoft/azuredatastudio/blob/main/extensions/arc/src/test/models/controllerModel.test.ts#L111